File: Question/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;  // val pos

        // build hashmap
        for (int i = 0; i < nums.size(); ++i) {
            map[nums[i]] = i;
        }

        for (int i = 0; i < nums.size(); ++i) {
            int less = target - nums[i];
            if (map.count(less) != 0 && map[less] != i) return {i, map[less]};
        }
        return {};
    }
};

File: Question_Java/1_Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mem = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (mem.containsKey(target - nums[i])) {
                return new int[]{i, mem.get(target - nums[i])};
            }
            mem.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }
}